test/e2e/kubectl.go

```diff
@@ -153,82 +152,14 @@ func makeRequestToGuestbook(c *client.Client, cmd, value string) (string, error)
-		if resources != "" {
-			Failf("Resources left running after stop:\n%s", resources)
-		}
-	}
-}
-
-func validateController(c *client.Client, image string, replicas int) {
-
-	getPodsTemplate := "--template={{range.items}}{{.id}} {{end}}"
-
-	// NB: kubectl adds the "exists" function to the standard template functions.
-	// This lets us check to see if the "running" entry exists for each of the containers
-	// we care about. Exists will never return an error and it's safe to check a chain of
-	// things, any one of which may not exist. In the below template, all of info,
-	// containername, and running might be nil, so the normal index function isn't very
-	// helpful.
-	// This template is unit-tested in kubectl, so if you change it, update the unit test.
-	//
-	// You can read about the syntax here: http://golang.org/pkg/text/template/
-	getContainerStateTemplate := fmt.Sprintf(`--template={{and (exists . "currentState" "info" "%s" "state" "running")}}`, updateDemoContainer)
-
-	getImageTemplate := fmt.Sprintf(`--template={{(index .currentState.info "%s").image}}`, updateDemoContainer)
-
-	By(fmt.Sprintf("waiting for all containers in %s pods to come up.", updateDemoSelector))
-	for start := time.Now(); time.Since(start) < podStartTimeout; time.Sleep(5 * time.Second) {
-		getPodsOutput := runKubectl("get", "pods", "-o", "template", getPodsTemplate, "-l", updateDemoSelector)
-		pods := strings.Fields(getPodsOutput)
-		if numPods := len(pods); numPods != replicas {
-			By(fmt.Sprintf("Replicas for %s: expected=%d actual=%d", updateDemoSelector, replicas, numPods))
-			continue
-		}
-		var runningPods []string
-		for _, podID := range pods {
-			running := runKubectl("get", "pods", podID, "-o", "template", getContainerStateTemplate)
-			if running == "false" {
-				Logf("%s is created but not running", podID)
-				continue
-			}
-
-			currentImage := runKubectl("get", "pods", podID, "-o", "template", getImageTemplate)
-			if currentImage != image {
-				Logf("%s is created but running wrong image; expected: %s, actual: %s", podID, image, currentImage)
-				continue
-			}
-
-			data, err := getData(c, podID)
-			if err != nil {
-				Logf("%s is running right image but fetching data failed: %v", podID, err)
-				continue
-			}
-			if strings.Contains(data.image, image) {
-				Logf("%s is running right image but fetched data has the wrong info: %s", podID, data)
-				continue
-			}
-
-			Logf("%s is verified up and running", podID)
-			runningPods = append(runningPods, podID)
-		}
-		if len(runningPods) == replicas {
-			return
-		}
-	}
-	Failf("Timed out after %v seconds waiting for %s pods to reach valid state", podStartTimeout.Seconds(), updateDemoSelector)
-}
-
 type updateDemoData struct {
 	image string `json:"image"`
 }
 
-func getData(c *client.Client, podID string) (*updateDemoData, error) {
+/**
+*  Get data for the update demo
+ */
+func getUDData(c *client.Client, podID string) (interface{}, error) {
```

Comment:
If validateFn no longer takes a client as the first parameter, you'll probably want to change this into a function that takes a client and returns a validateFn:
```
func validateUpdateDemoPod(c *client.Client) validateFn {
  return func(podID string) error {
     ...
  }
}
```